pkg/minishift/registration/registration.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"errors"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+var (
+	ErrDetectionFailed = errors.New("Distro type not recognized to Registration")
+	registrators       = make(map[string]*RegisteredRegistrator)
+	detector  Detector = &StandardRegistrator{}
+)
+
+
+type SSHCommander interface {
```

Comment:
@hferentschik here is only interface and actual implementation happen in the different registrator/registrar side.